plasmapy/diagnostics/thomson.py

```diff
@@ -84,50 +293,57 @@ def spectral_density(
         Wavelength of the probe laser. (convertible to nm)
 
     n : `~astropy.units.Quantity`
-        Mean (0th order) density of all plasma components combined.
-        (convertible to cm\ :sup:`-3`\ .)
+        Total combined number density of all electron populations.
+        (convertible to cm\ :sup:`-3`)
 
-    Te : `~astropy.units.Quantity`, shape (Ne, )
+    T_e : `~astropy.units.Quantity`, shape (Ne, )
         Temperature of each electron component. Shape (Ne, ) must be equal to the
-        number of electron components Ne. (in K or convertible to eV)
+        number of electron populations Ne. (in K or convertible to eV)
 
-    Ti : `~astropy.units.Quantity`, shape (Ni, )
+    T_i : `~astropy.units.Quantity`, shape (Ni, )
```

Comment:
Come to think of it, the change of `Te`, `Ti` → `T_e`, `T_i` probably warrants a `974.breaking.rst` changelog entry, since I think this was in the 0.7.0 release.

Reply:
I don't think it's breaking because these are positional arguments, so previous code should still work the same?

Reply:
It'd still be possible to provide them as keyword arguments, so if anyone had done something like `spectral_density(..., Te=, Ti=)` then they'd need to update their code.

Come to think of it, another possibility here would be to make `T_e` and `T_i` keyword-only arguments.  
```Python
def f(posarg, *, T_i, T_e):
```
My motivation is that it'd be easy to mix up `T_i` and `T_e` without knowing it, and requiring calls like `f(1, T_i = 5 * u.K, T_e = 6 * u.K)` would greatly reduce the chance of making that error.

Reply:
I agree with that, but they are required: are required keywords ok?

Reply:
That's a good question!  If you don't put in a default value, then the keyword-only arguments are required.  I forgot to mention that in the parameter type description, we'd need to include that it's keyword-only and/or optional.  Something like...
```Python
def f(*, T_i = 5 *u.K, T_e):  # T_i is optional, T_e is required
    """
    Parameters
    ----------
    T_i : ~astropy.units.Quantity, keyword-only, optional
        The ion temperature.  Defaults to 5 K.

    T_e : ~astropy.units.Quantity, keyword-only
        The electron temperature.
```

Reply:
Great, done!